_query-dsl/joining/has-parent.md

```diff
@@ -0,0 +1,358 @@
+}
+```
+
+For more information about retrieving inner hits, see [Inner hits]({{site.url}}{{site.baseurl}}/search-plugins/searching-data/inner-hits/).
+
+## Parameters
+
+The following table lists all top-level parameters supported by `has_parent` queries.
+
+| Parameter  | Required/Optional | Description  |
+|:---|:---|:---|
+| `parent_type` | Required | Specifies the name of the parent relationship as defined in the `join` field mapping. |
+| `query` | Required | The query to run on parent documents. If a parent document matches the query, the child document is returned. |
+| `ignore_unmapped` | Optional | Indicates whether to ignore unmapped `parent_type` fields and not return documents instead of throwing an error. You can provide this parameter when querying multiple indexes, some of which may not contain the `parent_type` field. Default is `false`. |
+| `score` | Optional | Indicates whether the relevance score of a matching parent document is aggregated into its child documents. If `false`, the relevance score of the parent document is ignored, and each child document is assigned a relevance score equal to the query's `boost`, which defaults to `1`. If `true`, the relevance score of the matching parent document is aggregated into the relevance scores of its child documents. Default is `false`. |
+| `inner_hits` | Optional | If provided, returns the underlying hits (parent documents) that matched the query. |
+
+
+## Sorting limitations
+
+The `has_parent` query does not support [sorting results]({{site.url}}{{site.baseurl}}/search-plugins/searching-data/sort/) using standard sorting options. If you need to sort child documents by fields in their parent documents, you can use a [`function_score` query]({{site.url}}{{site.baseurl}}/query-dsl/compound/function-score/) and sort by the child document's score. 
+
+For the preceding example, first add a `customer_satisfaction` field by which you'll be sorting the child documents to the parent (brand) documents:
+
+```json
+PUT testindex1/_doc/1
+{
+  "name": "Luxury watch brand",
+  "product_to_brand" : "brand",
+  "customer_satisfaction": 4.5
+}
+```
+{% include copy-curl.html %}
+
+```json
+PUT testindex1/_doc/2
+{
+  "name": "Economy watch brand",
+  "product_to_brand" : "brand",
+  "customer_satisfaction": 3.9
+}
+```
+{% include copy-curl.html %}
+
+Now you can sort child documents (products) based on the `customer_satisfaction` field of their parent brands. This query multiplies the score by the `customer_satisfaction` field of the parent documents:
+
+```json
+GET testindex1/_search
+{
+  "query": {
+    "has_parent": {
+      "parent_type": "brand",
+      "score": true,
+      "query": {
+        "function_score": {
+          "script_score": {
+            "script": "_score * doc['customer_satisfaction'].value"
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response contains the products sorted by the highest parent `customer_satisfaction`:
```

Comment:
Either a comma should follow "products", or "the" should be omitted before "products".